### DISCUSSION TITLE
Mac本地打包和Linux部署打包产物不同，chunk中出现React，导致React is not defined 报错

### DISCUSSION THREAD

**Discussion by GhostFJ** [umijs/umi#12371] [Q&A]
1. 环境：
本地：
- mbp m1 v13.2.1 (22D68)
- node ：v16.17.1

线上：
- Linux Qoffline101005 5.4.119-19-0009.1 #1 SMP Sun Jan 23 23:20:30 CST 2022 x86_64 x86_64 x86_64 GNU/Linux
- node：v16.20.2

项目：umi4 + antd pro
代码和配置都一样，直接git clone

结果：
线上直接打包 npm run build后访问，报错：
![image](https://github.com/umijs/umi/assets/35386856/307090c5-b2f2-4ebb-958e-7067ef3c1120)

![image](https://github.com/umijs/umi/assets/35386856/af80fe24-8c6f-4c55-8860-1207b08ed95d)

直接使用mac本地打包后的dist，正常访问

**Comment by fz6m** [comment]
你是开了生产时的 external 然后用的 umd 版本吗，看看是不是全局脚本没加载。

如果不是 external 的问题，看看是不是本地和服务器上构建装的依赖不一样，确保本地和云的 node 版本一样，包管理器版本一样，lock 文件一样，依赖都一样的情况下再构建。

如果还有问题，需要给一个最小复现。

**Reply by GhostFJ** [reply]
感谢，之前已经解决了，确实是版本依赖问题，linux安装时安装了最新的pro-components@2.7.2和react@18.3.1，所以导致报错
